sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/FileQuerySerialization.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.datalake.models;
+
+/**
+ * Defines the input and output serialization for a file query request.
+ * either {@link FileQueryJsonSerialization} or {@link FileQueryDelimitedSerialization}
+ */
+public abstract class FileQuerySerialization {
+
+    protected char recordSeparator;
+
+    /**
+     * Gets the record separator.
+     *
+     * @return the record separator.
+     */
+    public char getRecordSeparator() {
+        return recordSeparator;
+    }
+
+    abstract FileQuerySerialization setRecordSeparator(char recordSeparator);
```

Comment:
I think we should just have normal setter here. Derivatives shouldn't be concerned about this field - I assume we're using inheritance to share this property.

Reply:
I originally wanted to do that but the issue I had with that was we want to preserve the specific type to keep things  Fluent. Otherwise I wouldnt be able to do something like this to make a BlobQuerySerialization. 

blobQuerySerialization = new BlobQueryDelimitedSerialization().setRecordSeparator('\n').setColumnSep.....

Reply:
Take a look on this creative solution https://stackoverflow.com/questions/14105275/fluent-setters-with-inheritance-in-java .

Other option would be to move implementation here and just call `super.setRecordSeparator` + cast in derivatives to avoid repeating implementation.

Reply:
Yeah I've seen the one where you suppress warnings before, but I didnt like the fact that I would need to suppress warnings.

Reply:
done